monitor/linuxmonitor/cgnetcls/cgnetcls_test.go

```diff
@@ -24,13 +24,17 @@ const (
 	testmark             = 100
 )
 
-// TODO: remove nolint
-// nolint
-func cleanupnetclsgroup() {
+func cleanupnetclsgroup(err *error) {
```

Comment:
Why do you need to have the error here ?

Reply:
This function calls ```ioutil.WriteFile``` and ```os.RemoveAll``` . This is a called by ```defer cleanupnetclsgroup(&err)``` from other functions. So I am just passing the err pointer to the cleanupnetclsgroup to record the errors. Not sure why it defered though.

Reply:
This is useless

```golang
      defer cleanupnetclsgroup(&cerr)
      if cerr != nil {
 		t.Errorf("Failed to clean cgroup %s", cerr.Error())
      }	  
```

This piece of code will never be reached since it's refer. You should pass the pointer `t  *testing.T`. Then in this method, of there is an error you'll be able to make fail the test